test/jb_splitsStore/set.test.js

```diff
@@ -0,0 +1,291 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { deployMockContract } from '@ethereum-waffle/mock-contract';
+import jbOperatorStore from '../../artifacts/contracts/JBOperatorStore.sol/JBOperatorStore.json';
+import jbProjects from '../../artifacts/contracts/JBProjects.sol/JBProjects.json';
+import jbDirectory from '../../artifacts/contracts/JBDirectory.sol/JBDirectory.json';
+
+const ZERO_ADDRESS = ethers.constants.AddressZero; // address(0)
+const ONE_DAY = 3600000;
+
+describe.only('JBSplitsStore::set(...)', function () {
+  const PROJECT_ID = 1;
+  const DOMAIN = 2;
+  const GROUP = 3;
+  let SET_SPLITS_PERMISSION_INDEX;
+
+  before(async function () {
+    let jbOperationsFactory = await ethers.getContractFactory('JBOperations');
+    let jbOperations = await jbOperationsFactory.deploy();
+    SET_SPLITS_PERMISSION_INDEX = await jbOperations.SET_SPLITS();
+  });
+
+  async function setup() {
+    let [deployer, projectOwner, ...addrs] = await ethers.getSigners();
+
+    let mockOperatorStore = await deployMockContract(deployer, jbOperatorStore.abi);
+    let mockProjects = await deployMockContract(deployer, jbProjects.abi);
+    let mockDirectory = await deployMockContract(deployer, jbDirectory.abi);
+
+    await mockOperatorStore.mock.hasPermission
+      .withArgs(projectOwner.address, projectOwner.address, PROJECT_ID, SET_SPLITS_PERMISSION_INDEX)
+      .returns(true);
+
+    await mockProjects.mock.ownerOf
+      .withArgs(PROJECT_ID)
+      .returns(projectOwner.address);
+
+    await mockDirectory.mock.controllerOf
+      .withArgs(PROJECT_ID)
+      .returns(projectOwner.address);
+
+    let jbSplitsStoreFact = await ethers.getContractFactory('JBSplitsStore');
+    let jbSplitsStore = await jbSplitsStoreFact.deploy(
+      mockOperatorStore.address,
+      mockProjects.address,
+      mockDirectory.address
+    );
+
+    let splits = createSplitArray(addrs[0].address, 4);
+
+    return { deployer, projectOwner, addrs, jbSplitsStore, splits, mockOperatorStore, mockProjects, mockDirectory };
+  }
+
+  // Create array of JBSplit struct
+  function createSplitArray(beneficiaryAddress, n) {
```

Comment:
consider renaming to makeSplits

Reply:
also consider swapping n with 'count' and having it default to 4, which seems commonly used.

function makeSplits(beneficiaryAddress, count = 4) {
...
}